CandlestickDAOChartCView

kennyqiyenkan opened the pull request:
A chart that is created from chartjs using the data inside a CandlestickDAO.

kennyqiyenkan committed:
prelim general CandlestickDAOChartCView

kennyqiyenkan committed:
removed overcomplicated thought process

kennyqiyenkan committed:
testing with mock data working

kennyqiyenkan committed:
cleaned up code and documented purpose and logic process

kennyqiyenkan committed:
uncommented workaround

mcarcaso commented on src/org/chartjs/CandlestickDAOChartCView.js:
```diff
@@ -0,0 +1,204 @@
+    },
+    {
+      class: 'Map',
+      name: 'candlestickMap',
```
Do you need this as a property?

mcarcaso commented on src/org/chartjs/CandlestickDAOChartCView.js:
```diff
@@ -0,0 +1,204 @@
+      var self = this;
+      this.add(this.slot(function( config ) {
+        if ( config ) {
+          return this.E().tag(self.ChartCView, { config: self.config });
```
self.config can be just config

mcarcaso commented on src/org/chartjs/CandlestickDAOChartCView.js:
```diff
@@ -0,0 +1,204 @@
+
+        var self = this;
+        // Get all candlesticks categorized by their key, and ordered in ascending order by their closeTime.
+        this.data.orderBy(this.Candlestick.CLOSE_TIME).select(this.GROUP_BY(this.Candlestick.KEY, this.ArraySink.create())).then( function(a) {
```
There's a bunch of jumping around here that makes this kinda tough to read.
generateConfig goes to generateData which goes to generateDataSets, etc.

Check out https://github.com/foam-framework/foam2/pull/2103 and the test that's in the comment. I think using that approach to get the dataset together can help clean this up a bunch and I think the entire config can be done in one method since I don't see us needing to reuse much of these methods.

kennyqiyenkan replied:
I personally find that to be pretty hard to read. I'll see what I can do simplify the functions

kennyqiyenkan committed:
addressed a comment

kennyqiyenkan committed:
addressed another comment

mcarcaso committed:
Merge branch 'PlotSink' into kenny/Charts

mcarcaso committed:
Simplify some of the chartjs code.

mcarcaso committed:
Fix up multiview's property to be an array rather than FObjectArray because we don't want created views.
Make the CandlestickDAOChartCView copy the config before modifying and re-setting it
Add a comment explaining the height/width on ChartCView.

mcarcaso committed:
Rename CandlestickDAOChartCView to CandlestickDAOChartView because it's not actually a CView, it's a view that has a CView as a child.

mcarcaso committed:
Merge with master
